internal/action/bufpane.go

```diff
@@ -166,8 +166,12 @@ func BufMapEvent(k Event, action string) {
 					break
 				}
 			}
-			// if the action changed the current pane, update the reference
-			h = MainTab().CurPane()
+			// if the action changed the current BufPane to a BufPane again, update the reference
+			bp, ok := MainTab().CurPane().(*BufPane)
+			if !ok {
+				return innerSuccess
```

Comment:
But `innerSuccess` is modified in the condition...
`innerSuccess = innerSuccess && h.execAction(a, names[i], j, te)`
...depending on the result of `execAction()` or do I oversee something?
In case we `break` we return the forced `true`.

Reply:
So, as I said, it doesn't look like this return value is used anywhere anyway. So it doesn't look like it makes sense to return anything else than the forced `true` here. (Or, if we still want to return `innerSuccess` here, then at least for consistency we should also at line 277 return `success` instead of the forced `true`, right?)

AFAICS `success` and `innerSuccess` are only for determining if we should execute the next action, but for determining which value should `bufAction` return.

I actually have no idea what is the semantics of this return value. To be precise, I actually see just one place where this return value is used: https://github.com/zyedidia/micro/blob/master/internal/action/infopane.go#L143

and, this particular code path looks VERY confusing to me. Can you explain to me what does this code do and why does it use this return value the way it does? Why doesn't InfoPane's `DoKeyEvent()` just do the same thing as BufPane's `DoKeyEvent()`?

Reply:
Reading your question again... Yes, `innerSuccess` is modified depending on the value returned by `execAction()`, but what does it have to do with the value returned by `bufAction`?

Reply:
> but what does it have to do with the value returned by `bufAction`?

Unfortunately nothing.

> [...] return `success` instead of the forced `true`, right?

Yes and so I did.

Reply:
Answering myself:

> To be precise, I actually see just one place where this return value is used: https://github.com/zyedidia/micro/blob/master/internal/action/infopane.go#L143
>
> and, this particular code path looks VERY confusing to me. Can you explain to me what does this code do and why does it use this return value the way it does? Why doesn't InfoPane's `DoKeyEvent()` just do the same thing as BufPane's `DoKeyEvent()`?

Ok, I've figured out myself what's up with that logic in InfoPane's `DoKeyEvent()`. That logic makes sense, although is quite non-obvious and also somewhat buggy... PR #3267 is the result of my investigations.

> > [...] return `success` instead of the forced `true`, right?
> 
> Yes and so I did.

And I'm still not sure it's the best idea. I still believe this return value is not really used, and should not really be used. (Please see https://github.com/zyedidia/micro/pull/3267/commits/8c7f63ac15862bd8ecafbcafc5ec45d6e1b00fef and let me know if you disagree with the commit message.)

Well... if we merge #3267, including https://github.com/zyedidia/micro/pull/3267/commits/8c7f63ac15862bd8ecafbcafc5ec45d6e1b00fef, then there will be no more code that (improperly) tries to use this return value. So then I'm ok with either keeping this `return true` here or changing it to `return success`.

Reply:
Just a nit though: it's better to do this (change to `return success`) in a separate commit, since it is a separate change, not related to the fix for the crash.